src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,123 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { NewCommentForm } from '../NewCommentForm';
+import { PostData } from '../../types/PostData';
+import { getPostDetails } from '../../api/posts';
+import { getPostComments, deleteComment } from '../../api/comments';
+import { Comment } from '../../types/Comment';
 import './PostDetails.scss';
 
-export const PostDetails: React.FC = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
+type Props = {
+  selectedPostId: number;
+};
 
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
+export const PostDetails: React.FC<Props> = ({ selectedPostId }) => {
+  const [singlePost, setSinglePost] = useState<PostData | null>(null);
+  const [comments, setComments] = useState<null | Comment[]>(null);
+  const [show, setShow] = useState<boolean>(true);
+  const [commentsNumber, setCommentsNumber]
+  = useState<number>(comments ? comments.length : 0);
 
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
+  useEffect(() => {
+    if (selectedPostId < 1) {
+      return;
+    }
 
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
+    getPostDetails(selectedPostId)
+      .then(res => {
+        if (res.responseError.error === null) {
+          setSinglePost(res.data);
+        } else {
+          // eslint-disable-next-line no-console
+          console.log(`${res.responseError.message}`);
+        }
```

Comment:
don't do this on front - make it at api folder and be sure that here data will be valid